### PR TITLE
Manually entered repository source

### PR DESCRIPTION
**Pull request by Vlatombe**
@reviewbybees 

Allows to enter manually a repository to notify.

This is useful when you are working with a pipeline cloning multiple repositories and you want to notify only one of them on a specific SHA.

---
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/github-plugin/131)

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySource/help-url.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div>
+    A GitHub repository URL.
+</div>
```
Comment (KostyaSha) [inline src/main/resources/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySource/help-url.html]: new line

File: src/main/resources/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySource/config.groovy [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.jenkinsci.plugins.github.status.sources.AnyDefinedRepositorySource
+
+
+def f = namespace(lib.FormTagLib);
+
+f.entry(title: _('Repository URL'), field: 'url') {
+    f.textbox()
+}
```
Comment (KostyaSha) [inline src/main/resources/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySource/config.groovy]: new line

**Comment by Vlatombe** [comment]
@KostyaSha added the missing newlines

**Comment by lanwen** [comment]
Please add a little test

**Comment by lanwen** [comment]
*[src/main/java/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySource.java, line 41 \[r2\]](https://reviewable.io:443/reviews/jenkinsci/github-plugin/131#-KNCHgS-95M0ENBJP4VX:-KNCHgS-95M0ENBJP4VY:2120809007) ([raw file](https://github.com/jenkinsci/github-plugin/blob/0e620aa9afd6499d88ecc20d32b36d5dd9938e62/src/main/java/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySource.java#L41)):*
> ```Java
>                     return name.resolve();
>                 } else {
>                     listener.getLogger().println("Unable to match " + url + " with a GitHub repository.");
> ```

logger can use formatting to avoid string concatenation via \+

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/jenkinsci/github-plugin/131)*

**Comment by lanwen** [comment]
Reviewed 1 of 4 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/jenkinsci/github-plugin/131)*

**Comment by Vlatombe** [comment]
I have added a test when url cannot be converted into a repository name. However adding more coverage looks complicated because would need Jenkins instance and a valid connection to GitHub to resolve to a GHRepository.

**Pull request merged**